### PR TITLE
apiv2(ticdc): health api

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref #8019

### What is changed and how it works?
implement health api v2

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/api/v2/health.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+func (h *OpenAPIV2) health(c *gin.Context) {
+	if !h.capture.IsOwner() {
+		middleware.ForwardToOwnerMiddleware(h.capture)(c)
+		return
+	}
+
+	ctx := c.Request.Context()
+	health, err := h.capture.StatusProvider().IsHealthy(ctx)
+	if err != nil {
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
```
Comment (asddongmen) [inline cdc/api/v2/health.go:34]:
```suggestion
		_ = c.Error(err)
```

File: cdc/api/v2/health.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		return
+	}
+	if !health {
+		err = cerror.ErrClusterIsUnhealthy.FastGenByArgs()
+		c.IndentedJSON(http.StatusInternalServerError, model.NewHTTPError(err))
```
Comment (asddongmen) [inline cdc/api/v2/health.go:39]: ditto

File: cdc/api/v2/health.go (hunk 1 above)
Comment (sdojjy) [reply cdc/api/v2/health.go]: done

File: cdc/api/v2/health.go (hunk 2 above)
Comment (sdojjy) [reply cdc/api/v2/health.go]: done

**Review by asddongmen — APPROVED** [review]

**Pull request merged**